### ISSUE TITLE
Replace interordinal internals with a single system

### ISSUE THREAD

**Issue by frostburn** [xenharmonic-devs/sonic-weave#297]
**pinkanberry** — _Today at 8:43 PM_

> table should be complete, the 1-indexed notation to use would be 3x4 and 5x6 if you are okay with that ⁠notation⁠
> P3.5 is definitely not perfect (unless you want to increase the range to three generators but then theres even more possibilities to consider), but logically it would be the semisixth of sqrt(27/16)

> 
> P0dias = 1/1 = 0g
> A0dias = 2187/2048
> P0.5dias = sqrt(9/8) = 1p - 2g = 0x1dias
> m1dias = 256/243
> n1dias = sqrt(32/27)
> M1dias = 9/8
> P1.5dias = sqrt(4/3) = 1g = 1x2dias
> m2dias = 32/27
> N2dias = sqrt(3/2) = 1p - 1g
> M2dias = 81/64
> 2x3dias = sqrt(27/16) = sqrt(27/8) / sqrt(2) = 2p - 3g
> P3dias = 4/3 = 2g
> n3dias = 11/8 (tempering 243/242)
> A3dias = 729/512
> P3.5dias = sqrt(2) = 1p
> d4dias = 1024/729
> n4dias = 16/11 (tempering 243/242)
> P4dias = 3/2 = 2p - 2g
> 3x4dias = 2 / sqrt(27/16) = sqrt(64/27) = 3g
> m5dias = 128/81
> N5dias = 2 / sqrt(3/2) = sqrt(8/3) = 1p + 1g
> M5dias = 27/16
> P5.5dias = 2/sqrt(4/3) = sqrt(3) = 2p - 1g = 5x6dias
> m6dias = 16/9
> n6dias = 2 / sqrt(32/27) = sqrt(27/8)
> M6dias = 243/128
> P6.5dias = 2 / sqrt(9/8) = sqrt(32/9) = 1p + 2g = 6x7dias
> d7dias = 4096/2187
> P7dias = 2/1 = 2g
> 

The table above should be adapted with a fist sized grain of salt.

**Comment by osmiumic** [comment]
please refer to my responses here: https://github.com/xenharmonic-devs/sonic-weave/issues/298

i've posted the 1-indexed and hopefully correct table [here](<https://github.com/xenharmonic-devs/sonic-weave/issues/298#issuecomment-2094926427 https://github.com/xenharmonic-devs/sonic-weave/issues/298>) in that issue

**Comment by osmiumic** [comment]
also this feels potentially very relevant to this issue so i'll repost it here:
> importantly: semi-augmentation corresponds to moving (upwards) by (3p - 5g) - 2g = 3p - 7g (so augmentation is therefore the immensely complex 6p - 14g)

**Comment by osmiumic** [comment]
one more thing: really important:

using 10L 4s has a really nice consequence:

ABCDEFG can keep their normal musical meanings, and you get 7 mathematically/structurally exact greek analogues:
alpha, beta, gamma, delta, epsilon, zeta, omega

(i suggest using omega, the last letter, instead of eta because eta sounds like it would be E)

these are *exactly* off from the normal chain by sqrt(4/3) (by the generator; not by the period cuz remember the period is always available everywhere!)

in other words, [alpha beta gamma delta epsilon zeta omega] is the same as [A B C D E F G] as pyth[7] offset by sqrt(4/3)! i don't think you can get more elegant than that ^_^

altho: i suggest considering a way of shortening the greek letter names to either all be 2 letters long or 3 letters long, and i also strongly recommend allowing the first letter to be capitalised for consistency with ABCDEFG

**Comment by frostburn** [comment]
> (i suggest using omega, the last letter, instead of eta because eta sounds like it would be E)

Disagree. By the same token beta sounds like B-ta and delta sounds like D-lta depending on your pronunciation.

> in other words, [alpha beta gamma delta epsilon zeta omega] is the same as [A B C D E F G] as pyth[7] offset by sqrt(4/3)! i don't think you can get more elegant than that ^_^

Disagree. The current system is more elegant and (anti-)symmetric in the span of the octave. I would also like to point out that even equal temperaments like 22 are more in need of alternative notation than rarer ETs that split the fourth. (That said I will try to marry relative and absolute notation carefully so we might get this inelegant offset anyway. Don't see why we would use the generator when we can use the period for offset, though.)

**Comment by osmiumic** [comment]
> Don't see why we would use the generator when we can use the period for offset, though.

because offsetting by the period would result in missing notes; fundamentally there is two pyth[7] chains offset by sqrt(4/3) or equivalently by sqrt(3/2) (since sqrt(4/3) * sqrt(3/2) = sqrt(2)). there's no way around this if you want to preserve any semblance of 10L 4s logic because the way we are getting pyth[7] is by taking even numbers of generators where the generator is either sqrt(4/3) or sqrt(3/2). therefore fixing ABCDEFG to pyth[7] naturally implies that you only have 7 notes left available in the natural scale of 10L 4s. it is a *14*-note natural scale, not a 12-note one.

> By the same token beta sounds like B-ta and delta sounds like D-lta depending on your pronunciation.

isnt that a good thing? A=Alpha, B=Beta, C=Gamma, D=Delta, E=Epsilon, F=Zeta, G=Eta i find confusing because i'd easily confuse epsilon and eta; contrast with the fact that omega has a G in it; i think most people are aware of the first five, F=Zeta tends to be rarer and a bit more confusing, so already we have that cuz it's starting with "Z" it sounds like it should be the last one potentially for anyone not familiar with it, hence why i suggest "omega" as the one after it to make it clear that "omega" is clearly the last, like in "alpha and omega"

**Comment by frostburn** [comment]
Let's see here.

Notation **must** be extended in priority order.

### Step 0
Humanity invents 1-based ordinals. All is lost. May Spoob have mercy on our 0-diasteps!

### Unison and octave
P1 = 1/1
P8 = 2/1
P15 = P8 + P8
...

C4 = (reference frequency)
C3 = C4 - P8
C5 = C4 + P8
...

### Pythagoras
P5 = 3/2
P4 = P8 - P5
Minor and major intervals induced by 5L2s.
Diminished and augmented qualities have different semantics on perfect and imperfect classes.

D4 = C4 + M2
E4 = C4 + M3
F4 = C4 + P4
...

♯ = Aug1
b = dim1

### Neutral chain of fifths
sa1 = Aug1 / 2
sd1 = -sa1

n5 = P5 - sa1
n2 = (n5 + P5) mod P8
n6 = (n2 + P5) mod P8
n3 = (n6 + P5) mod P8
n7 = (n3 + P5) mod P8
n4 = (n7 + P5) mod P8

n1 = syntax error

#### Accidentals
t = sa1
d = sd1

### Semioctave
P4.5 = P8 / 2

#### Intervals keep their quality when shifted by a semioctave
n1.5 = (P4.5 + n5) mod P8
P1.5 = (P4.5 + P5) mod P8

n2.5 = (P4.5 + n6) mod P8
m2.5 = n2.5 + sd1
M2.5 = n2.5 + sa1

n3.5 = (P4.5 + n7) mod P8
m3.5 = n3.5 + sd1
M3.5 = n3.5 + sa1

sd4.5 = P4.5 + sd1
sa4.5 = P4.5 + sa1

n5.5 = (P4.5 + n2) mod P8
m5.5 = n5.5 + sd1
M5.5 = n5.5 + sa1

n6.5 = (P4.5 + n3) mod P8
m6.5 = n6.5 + sd1
M6.5 = n6.5 + sa1

P7.5 = (P4.5 + P4) mod P8
n7.5 = (P4.5 + n4) mod P8

#### Nominals
zeta4 = C4 + P4.5

eta4 = zeta4 + M2
alpha4 = zeta4 + M3
beta5 = zeta4 + P4
gamma5 = zeta4 + P5
delta5 = zeta4 + M6
epsilon5 = zeta4 + M7

## Conclusion
We have expressions for the important stuff.
P5 / 2 = n3
P8 / 2 = P4.5
P4 / 2 = n2.5 = P4.5 - n3

Latin nominals (#, b) let us access the chain of fifths starting from C4.
Latin nominals (t, d) let us access the chain of fifths starting from C4 + P5 / 2
Greek nominals (#, b) let us access the chain of fifths starting from C4 + P8 / 2
Greek nominals (t, d) let us access the chain of fifths starting from C4 + P4 / 2

**Comment by frostburn** [comment]
Based on Discord discussions my counterproposal should be implementable with the change that `A4 + P8 / 2 = alp5`.
